### ISSUE TITLE
Implicit query parameter fails with NPE on != null comparison in QueryParserV3.parseTerm()

### ISSUE THREAD

**Issue by slavap** [tzaeschke/zoodb#100]
Not sure, may be it's supposed behavior, but looks strange to me.

Test attached, rename png to zip.
![zoodbbugs2](https://cloud.githubusercontent.com/assets/4129763/26343996/efd5ed0e-3f52-11e7-90b6-5c22ad43bcca.png)

**Comment by tzaeschke** [comment]
Yes, this is a bug. It is pretty much known because the current query parser usually fails if the left hand side of a term is not a field (or if the right hand side is a field). This should be fixed with the new query engine.
